test/nodes/test_sampler.py

```diff
@@ -0,0 +1,94 @@
+        meta={"name": "1"},
+        id="3",
+    ),
+    Document(
+        content="""The Dothraki vocabulary was created by David J. Peterson well in advance of the adaptation. HBO hired the Language Creatio""",
+        meta={"name": "2"},
+        id="4",
+    ),
+    Document(
+        content="""The title of the episode refers to the Great Sept of Baelor, the main religious building in King's Landing, where the episode's pivotal scene takes place. In the world created by George R. R. Martin""",
+        meta={},
+        id="5",
+    ),
+]
+
+
+def test_top_p_sampling(top_p_sampler):
+    query = "What is the most important religious learning from the Bible?"
+    results = top_p_sampler.predict(query=query, documents=docs, top_p=0.98)
+    assert len(results) == 2
+    assert results[0].id == "1"
+    assert results[1].id == "5"
+
+
+def test_top_p_sampling_at_least_one_result(top_p_sampler):
+    query = "What is the most important building in King's Landing that has a religious background?"
+    results = top_p_sampler.predict(query=query, documents=docs, top_p=0.9)
+
+    # The top_p sampler should always return at least one result if strict is False (default)
+    # in this case the result with the highest prob score is returned even if top p is too low
+    assert len(results) == 1
+    assert results[0].id == "5"
+
+    # unless strict is set to True
+    sampler = TopPSampler(strict=True)
+    results = sampler.predict(query=query, documents=docs, top_p=0.9)
+    assert len(results) == 0
+
+
+@pytest.mark.integration
+@pytest.mark.skipif(
+    not os.environ.get("SERPERDEV_API_KEY", None),
+    reason="Please export an env var called SERPERDEV_API_KEY containing the serper.dev API key to run this test.",
+)
+def test_sampler_pipeline():
+    ws = WebSearch(api_key=os.environ.get("SERPERDEV_API_KEY", None))
+    sampler = TopPSampler()
+    pipe = Pipeline()
+    pipe.add_node(component=ws, name="ws", inputs=["Query"])
+    pipe.add_node(component=sampler, name="sampler", inputs=["ws"])
+    result = pipe.run(query="Who is the boyfriend of Olivia Wilde?")
+    assert "documents" in result
+    assert len(result["documents"]) > 0
+    assert isinstance(result["documents"][0], Document)
+
+
+@pytest.mark.integration
+@pytest.mark.skipif(
+    not os.environ.get("SERPERDEV_API_KEY", None),
+    reason="Please export an env var called SERPERDEV_API_KEY containing the serper.dev API key to run this test.",
+)
+def test_sampler_with_ranker_pipeline():
+    ws = WebSearch(api_key=os.environ.get("SERPERDEV_API_KEY", None))
+    sampler = TopPSampler()
+    ranker = SentenceTransformersRanker(model_name_or_path="cross-encoder/ms-marco-MiniLM-L-6-v2", top_k=2)
```

Comment:
You could use the existing ranker fixture here, it uses cross-encoder/ms-marco-MiniLM-L-12-v2